ledger/acctonline.go

```diff
@@ -621,104 +621,125 @@ func (ao *onlineAccounts) lookupOnlineAccountData(rnd basics.Round, addr basics.
 	var rewardsLevel uint64
 	var persistedData persistedOnlineAccountData
 
-	offset, err = ao.roundOffset(rnd)
-	inHistory := false
-	if err != nil {
-		var roundOffsetError *RoundOffsetError
-		if !errors.As(err, &roundOffsetError) {
+	for {
```

Comment:
The loop makes the code much more difficult I think. If the queried data is too old, let's just discard it.

Reply:
We need to maintain the cache.